java-client/session/src/main/java/io/deephaven/client/impl/ExportStates.java

```diff
@@ -133,19 +135,39 @@ synchronized List<Export> export(ExportsRequest requests) {
             if (request.getOpsCount() == 0) {
                 throw new IllegalStateException();
             }
+            final BatchHandler batchHandler = new BatchHandler(newStates);
+            send = () -> tableStub.batch(request, batchHandler);
+        } else {
+            send = () -> {
+            };
+        }
+        return new ExportServiceRequest() {
 
-            // log.info("Sending batch: {}", request);
+            @Override
+            public List<Export> exports() {
+                return results;
+            }
 
-            tableStub.batch(request, new BatchHandler(newStates));
-            ++batchCount;
-        }
+            @Override
+            public void send() {
+                send.run();
+            }
 
-        return results;
+            @Override
+            public void close() {
```

Comment:
We have the potential for state leak if the user calls close without calling send. Do we need to clean those up? What about on errors?

Reply:
Good catch. I've added some new logic to handle the case where the user does not call send. After sent, any errors are the handled by respective `io.deephaven.client.impl.ExportRequest.listener`.